discovery-provider/src/queries/queries.py

```diff
@@ -2063,30 +2073,46 @@ def get_remixes_of(track_id):
                     Repost.is_current == True,
                     Repost.is_delete == False
                 )
-            ).filter(
+            ).outerjoin(
+                repost_count_subquery,
+                repost_count_subquery.c['id'] == Track.track_id
+            ).outerjoin(
+                save_count_subquery,
+                save_count_subquery.c['id'] == Track.track_id
+            )
+            .filter(
                 Track.is_current == True,
                 Track.is_delete == False,
                 Track.is_unlisted == False
             )
+            # 1. Co-signed tracks ordered by save + repost count
+            # 2. Other tracks ordered by save + repost count
             .order_by(
                 desc(
-                    case([
-                        (and_(Repost.created_at == None, Save.created_at == None), 0),
-                    ],
-                    else_ = 1)
+                    # If there is no "co-sign" for the track (no repost or save from the parent owner),
+                    # defer to secondary sort
+                    case(
+                        [
+                            (and_(Repost.created_at == None, Save.created_at == None), 0),
+                        ],
+                        else_ = (
+                            func.coalesce(repost_count_subquery.c.repost_count, 0) + \
+                            func.coalesce(save_count_subquery.c.save_count, 0)
+                        )
+                    )
                 ),
+                # Order by saves + reposts
                 desc(
-                    case([
-                        (or_(Repost.created_at == None, Repost.created_at <= Save.created_at), Save.created_at),
-                        (or_(Save.created_at == None, Repost.created_at >= Save.created_at), Repost.created_at)
-                    ],
-                    else_ = None)
+                    func.coalesce(repost_count_subquery.c.repost_count, 0) + \
```

Comment:
question about coalesce. it returns the first non null value or returns 0. but if everything is 0 here, would it ever become ORDER BY 0 DESC?

Reply:
yes -- but then it would fall back to track id, so it should be fine!

Reply:
just to confirm, the sql it outputs is not ORDER BY 0 DESC, 0 DESC, "track_id" DESC but rather ORDER BY "track_id" DESC, cause the former would cause problems if everything was null